plugins/CoreHome/javascripts/top_controls.js

```diff
@@ -108,3 +108,34 @@ function blockPropegation(){
         e.stopPropagation();
     })
 }
+
+//refresh page short cut 'r'
+$(function () {
+  function refresh() {
+    var Matomo = window.CoreHome.Matomo;
+    var hashParsed = window.CoreHome.MatomoUrl.hashParsed.value;
+
+    Matomo.postEvent('loadPage', hashParsed.category, hashParsed.subcategory);
+  };
+
+  function isCoreHomeModuleActive() {
+    var search = window.CoreHome.MatomoUrl.parse(window.location.search.slice(1));
+    return search.module === 'CoreHome';
+  }
+
+  if (isCoreHomeModuleActive()) {
+
+    piwikHelper.registerShortcut('r', _pk_translate('CoreHome_ShortcutRefresh'), function (event) {
+      if (event.altKey) {
+        return;
+      }
+      if (event.preventDefault) {
+        event.preventDefault();
+      } else {
+        event.returnValue = false; // IE
+      }
+
+      refresh();
+    });
+  }
```

Comment:
I guess this could be simplified as the CoreHome check shouldn't be needed. Haven't tested it, but something like this should be enough:
```suggestion
  piwikHelper.registerShortcut('r', _pk_translate('CoreHome_ShortcutRefresh'), function (event) {
    if (event.altKey) {
      return;
    }
    if (event.preventDefault) {
      event.preventDefault();
    } else {
      event.returnValue = false; // IE
    }

    var Matomo = window.CoreHome.Matomo;
    var hashParsed = window.CoreHome.MatomoUrl.hashParsed.value;

    Matomo.postEvent('loadPage', hashParsed.category, hashParsed.subcategory);
  });
```